xsensmt/XsensMT.cpp

```diff
@@ -322,3 +325,195 @@ bool XsensMT::setOptionFlags(const uint32_t setFlags, const uint32_t clearFlags)
+    timestamp = copyStamp.getTime();
+    return true;
+}
+
+
+size_t XsensMT::getNrOfThreeAxisLinearAccelerometers() const
+{
+    return 1;
+}
+
+yarp::dev::MAS_status XsensMT::getThreeAxisLinearAccelerometerStatus(size_t sens_index) const
+{
+    return genericGetStatus(sens_index);
+}
+
+bool XsensMT::getThreeAxisLinearAccelerometerName(size_t sens_index, std::string& name) const
+{
+    return genericGetSensorName(sens_index, name);
+}
+
+bool XsensMT::getThreeAxisLinearAccelerometerFrameName(size_t sens_index, std::string& frameName) const
+{
+    return genericGetFrameName(sens_index, frameName);
+}
+
+bool XsensMT::getThreeAxisLinearAccelerometerMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp) const
+{
+    if (sens_index != 0 || m_isSensorMeasurementAvailable == false)
+    {
+        return false;
+    }
+    
+    out.clear();
+    out.resize(3);
+    std::lock_guard<std::mutex> guard(m_bufferMutex);
+    out[0] = m_sensorBuffer[3];
+    out[1] = m_sensorBuffer[4];
+    out[2] = m_sensorBuffer[5];
+    yarp::os::Stamp copyStamp(m_lastStamp);
+    timestamp = copyStamp.getTime();
+    return true;
+}
+
+
+size_t XsensMT::getNrOfThreeAxisGyroscopes() const
+{
+    return 1;
+}
+
+yarp::dev::MAS_status XsensMT::getThreeAxisGyroscopeStatus(size_t sens_index) const
+{
+    return genericGetStatus(sens_index);
+}
+
+bool XsensMT::getThreeAxisGyroscopeName(size_t sens_index, std::string& name) const
+{
+    return genericGetSensorName(sens_index, name);
+}
+
+
+bool XsensMT::getThreeAxisGyroscopeFrameName(size_t sens_index, std::string& frameName) const
+{
+    return genericGetFrameName(sens_index, frameName);
+}
+
+bool XsensMT::getThreeAxisGyroscopeMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp) const
+{
+    if (sens_index != 0 || m_isSensorMeasurementAvailable == false)
+    {
+        return false;
+    }
+    
+    out.clear();
+    out.resize(3);    
+    std::lock_guard<std::mutex> guard(m_bufferMutex);
+    out[0] = m_sensorBuffer[6];
+    out[1] = m_sensorBuffer[7];
+    out[2] = m_sensorBuffer[8];
+    yarp::os::Stamp copyStamp(m_lastStamp);
+    timestamp = copyStamp.getTime();
+    return true;
+}
+
+
+
+size_t XsensMT::getNrOfThreeAxisMagnetometers() const
+{
+    return 1;
+}
+
+
+yarp::dev::MAS_status XsensMT::getThreeAxisMagnetometerStatus(size_t sens_index) const
+{
+    return genericGetStatus(sens_index);
+}
+
+bool XsensMT::getThreeAxisMagnetometerName(size_t sens_index, std::string& name) const
+{
+    return genericGetSensorName(sens_index, name);
+}
+
+bool XsensMT::getThreeAxisMagnetometerFrameName(size_t sens_index, std::string& frameName) const
+{
+    return genericGetFrameName(sens_index, frameName);
+}
+
+bool XsensMT::getThreeAxisMagnetometerMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp) const
+{
+    if (sens_index != 0 || m_isSensorMeasurementAvailable == false)
+    {
+        return false;
+    }
+    
+    out.clear();
```

Comment:
No need to do clear: anything that is in the vector after the `resize(3)` will be overwritten.

Reply:
Fixed. Thanks!